src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -151,10 +165,33 @@ CHIP_ERROR BLEManagerImpl::_Init()
 
     memset(mSubscribedConns, 0, sizeof(mSubscribedConns));
 
-    ReturnErrorOnFailure(InitRandomStaticAddress());
-    int err = bt_enable(NULL);
+#ifdef CONFIG_BT_BONDABLE
+    static_assert(CONFIG_BT_ID_MAX == 2);
+
+    bt_addr_le_t idsAddr[CONFIG_BT_ID_MAX];
+    size_t idsCount = CONFIG_BT_ID_MAX;
+
+    err = bt_enable(nullptr);
+
     VerifyOrReturnError(err == 0, MapErrorZephyr(err));
 
+    settings_load();
+
+    bt_id_get(idsAddr, &idsCount);
+
+    err = InitRandomStaticAddress(idsCount > 1 ? true : false, id);
```

Comment:
```suggestion
    err = InitRandomStaticAddress(idsCount > 1, id);
```
comparison already returns bool